lib/utils.ts

```diff
@@ -1,39 +1,43 @@
-import {browser, ElementFinder} from "protractor";
-import {SaveFullPageMethodOptions} from "webdriver-image-comparison/build/commands/fullPage.interfaces";
+import { ElementFinder, ProtractorBrowser } from 'protractor';
+import { Folders } from 'webdriver-image-comparison/build/base.interface';
 
+import { InstanceData, BaseMethodOptions } from './interfaces';
+
+
+declare let browser: ProtractorBrowser;
 
 /**
  * Get the instance data
  */
-export async function getInstanceData() {
-	const instanceConfig = (await browser.getProcessedConfig()).capabilities;
+export async function getInstanceData(): Promise<InstanceData> {
+	const { capabilities = {} } = await browser.getProcessedConfig();
+	const options = [
+		'browserName',
+		'browserVersion',
+		'logName',
+		'platformName',
+		'deviceName'
+	];
 
-	// Substract the needed data from the running instance
-	const browserName = (instanceConfig.browserName || '').toLowerCase();
-	const browserVersion = instanceConfig.browserVersion || '';
-	const logName = instanceConfig.logName || '';
-	const name = instanceConfig.name || '';
+	const data = options.reduce<{ [key: string]: string }>((result, key) => {
+		const value: string = capabilities[key] || '';
+		result[key] = value.toLowerCase();
+		return result;
+	}, {});
 
-	// For mobile
-	const platformName = (instanceConfig.platformName || '').toLowerCase();
-	const deviceName = (instanceConfig.deviceName || '').toLowerCase();
-	const nativeWebScreenshot = !!instanceConfig.nativeWebScreenshot;
+	const nativeWebScreenshot = Boolean(capabilities.nativeWebScreenshot);
 
-	return {
-		browserName,
-		browserVersion,
-		deviceName,
-		logName,
-		name,
-		nativeWebScreenshot,
-		platformName,
-	};
+	return { ...data, nativeWebScreenshot };
 }
 
 /**
  * Transform all `hideElements`, `removeElements` and `hideAfterFirstScroll`-elements to WebElements
  */
-export async function optionElementsToWebElements(options: SaveFullPageMethodOptions) {
+export async function optionElementsToWebElements(options: BaseMethodOptions) {
+	// TODO (vitalie-ly): It's bad solution to mutate variable.
```

Comment:
Do you have a better solution for this then?

Reply:
We can do somethink like this
```typescript
const result = {  ...options };
result.hideElements = await getWebElements(options.hideElements);
// other stuff
return result;
```

By the way, we can change it later